modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ProcessorChainValueResolver.java

```diff
@@ -71,16 +74,9 @@ public boolean isDynamic() {
     return false;
   }
 
-  private void initialiseIfNeeded() {
-    if (!initialised.get()) {
-      try {
-        registerObject(muleContext, new ObjectNameHelper(muleContext).getUniqueName(""), chain);
-        LifecycleUtils.initialiseIfNeeded(chain, muleContext);
-      } catch (Exception e) {
-        throw new MuleRuntimeException(createStaticMessage("Could not register nested operation message processor"), e);
-      }
-      initialised.set(true);
-    }
+  @Override
+  public void initialise() throws InitialisationException {
+    initialiseIfNeeded(chain, muleContext);
```

Comment:
it doesn't make sense to do this here if the chain was registered before the initialisation phase (unless the constructor is invoked during the initialisation phase). All of this is very shady either way. Review, fix and/or clarify